selenium/codenvy-selenium-test/src/test/java/com/codenvy/selenium/factory/AuthenticateAndAcceptFactoryThroughGitHubOAuthTest.java

```diff
@@ -107,4 +111,43 @@ public void loginThroughGitHubOAuthAndAcceptFactory() throws Exception {
     projectExplorer.waitProjectExplorer();
     notificationsPopupPanel.waitExpectedMessageOnProgressPanelAndClosed("Project Spring imported");
   }
+
+  private void deleteAllWorkspaces(List<String> workspaces) {
+    workspaces.forEach(
+        wsName -> {
+          seleniumWebDriver.get(getDashboardWorkspaceUrl());
+          dashboardWorkspace.waitToolbarTitleName(WORKSPACE_TOOLBAR);
+          dashboardWorkspace.waitListWorkspacesOnDashboard();
+          dashboardWorkspace.selectWorkspaceItemName(wsName);
+          dashboardWorkspace.waitToolbarTitleName(Arrays.asList(wsName.split("/")).get(1));
+          dashboardWorkspace.clickOnDeleteWorkspace();
+          dashboardWorkspace.clickOnDeleteDialogButton();
+          dashboardWorkspace.waitToolbarTitleName(WORKSPACE_TOOLBAR);
+        });
+  }
+
+  private List<String> getAllWorkspaceNames() {
+    List<String> workspaces = new ArrayList<>();
+    getNotFilteredWorkspaceNames()
+        .forEach(
+            workspaceName -> {
+              if (isWorkspaceName(workspaceName)) {
+                workspaces.add(workspaceName);
+              }
+            });
+
+    return workspaces;
+  }
+
+  private List<String> getNotFilteredWorkspaceNames() {
+    return Arrays.asList(dashboardWorkspace.getTextFromListWorkspaces().split("\n"));
+  }
+
+  private boolean isWorkspaceName(String workspaceName) {
+    return workspaceName.contains("/") && workspaceName.length() > 3;
+  }
+
+  private String getDashboardWorkspaceUrl() {
```

Comment:
see comment here https://github.com/codenvy/codenvy/pull/2498/files#r145934139

Reply:
On my opinion, it's wrong way, we have necessary methods for work with this page. And methods which were created in the test are complicated for page object. It may be more difficult for debugging and support

Reply:
1. Logic of removing of workspaces from dashboard is not a propose of this class named **AuthenticateAndAcceptFactoryThroughGitHubOAuthTest**.
1. Test classes should be the easiest ones for debugging and support.
1. Some tests might require the same functionality and in this case we will need to duplicate the code.
1. This test could be removed in future and in this case we will lose the code.

Reply:
Yes, your arguments make sense, but it`s only one side of coin
1) PageObject must be as simple as possible
2) Test class will be easy to debug because if something will be wrong we just see it in the test class. So, i think, In this case, we can apply some code duplication.
3) Besides this methods were tested only for this tests, and i can`t give 100 % that they will be work correctly in all cases. So, at list, this is another separate task.

Reply:
Please, take a look https://en.wikipedia.org/wiki/Single_responsibility_principle

Reply:
Ok, I can create issue for testing and move this methods to page object, will you apply this?

Reply:
It's better to move the method to **DashboardWorkspace** page as part of the current issue.

Reply:
As I saz, I can`t give 100 % for all cases. This is another task that can take half of day. We need extra story points for this. I think the new task will be the better way.